lib/api/core/validation/index.js

```diff
@@ -452,11 +452,15 @@ Validation.prototype.curateCollectionSpecification = function validationCurateCo
     if (collectionSpec.fields) {
       try {
         result = this.structureCollectionValidation(collectionSpec, indexName, collectionName, verboseErrors);
-        if (verboseErrors && result.isValid === false) {
+        if (result.isValid === false) {
+          if (verboseErrors) {
+            // do not fail fast if we need verbose errors
+            return resolve(result);
+          }
           let error = new BadRequestError(result.errors[0]);
```

Comment:
It's considered good practice to declare a variable at the beginning of its scope. 
Furthermore:
  * either this `error` variable is unused and should be deleted, and there is a problem because you're rejecting a promise with a non-error object (should print a warning during unit tests if this code is covered)
  * or the promise should be rejected with the error object and the change made on the promise rejection should be reverted